### PR TITLE
fix: (Core) hide table card border, fix KPI value and correct card example with bottom header

### LINKED ISSUE #3928: Bug: Extra padding or spacing on Table Card not aligned with Visual Core
#### Is this a bug, enhancement, or feature request?
Bug

#### Briefly describe your proposal.
Extra padding/spacing on Table Card not aligned with Visual Core. Width of rows and headers on table should extend edge-to-edge. See spec below from Visual core:
image

#### Which versions of Angular and Fundamental Library for Angular are affected? (If this is a feature request, use current version.)
v0.24.0

#### If this is a bug, please provide steps for reproducing it.
1. Go to: https://sap.github.io/fundamental-ngx/#/core/card
2. Scroll down to: Table Card
3. See error: pic attached below
image

### LINKED ISSUE #4243: Fixed card layout: Color of the KPI value is wrong
**Describe the bug**

If we want to present a positive state, the KPI values should be green; negative state - red; neutral - grey.
But we don't use grey semantic color for KPI value and referring to a positive state. 
Fiori ref: 
https://ux.wdf.sap.corp/fiori-design-web/analytical-card/

### LINKED ISSUE #4710: Incorrect standard card example (card header at the bottom)
#### Is this a bug, enhancement, or feature request?
Enhancement

#### Briefly describe your proposal.
Please display an image instead of lists in the content area of an existing example of a standard card with the header at the bottom (shown below).
Screen Shot 2021-02-17 at 4 43 02 PM

#### Which versions of Angular and Fundamental Library for Angular are affected? (If this is a feature request, use current version.)
0.27.1-rc.5

#### Is there anything else we should know?
Please refer to the following examples.
Screen Shot 2021-02-17 at 4 58 07 PM

Additional examples in Figma:
https://www.figma.com/file/PrHXo4hVWfOdNgeo4WTHRi/Ariba-Component-Desktop?node-id=1078%3A2264

For more information, please see the Fiori3 Component Card specification:
https://wiki.wdf.sap.corp/wiki/display/visualcore/Card+%28Fiori+3%29+-+Component+Card#Card(Fiori3)ComponentCard-GeneralSpecification

### PR DESCRIPTION
**Pull request by DeepakSap14**
#### Please provide a link to the associated issue.
closes #3928
closes #4710
closes #4243

#### Please provide a brief summary of this pull request.
This PR creates card example required by design team

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [x] Run npm run build-pack-library and test in external application

Documentation checklist:
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

File: apps/docs/src/app/core/component-docs/card/examples/bar-chart-list-card/card-bar.component.ts
Comment (JKMarkowski) [inline apps/docs/src/app/core/component-docs/card/examples/bar-chart-list-card/card-bar.component.ts:22]: Is it needed? If so you could you change `displayChange: false` to `displayChange = false`?

File: apps/docs/src/app/core/component-docs/card/card-docs.component.html
Comment (JKMarkowski) [inline apps/docs/src/app/core/component-docs/card/card-docs.component.html:149]: Could you add `fd-docs-section-title` component, instead of `description`?

File: libs/core/src/lib/card/card.component.scss
Comment (InnaAtanasova) [inline libs/core/src/lib/card/card.component.scss:18]: we can't have hard-coded colors like `#ccc`.

File: libs/core/src/lib/card/card.component.scss
Comment (InnaAtanasova) [inline libs/core/src/lib/card/card.component.scss:22]: Is this css brought in Fundamental-styles?

File: apps/docs/src/app/core/component-docs/card/examples/list-card/card-list-example.component.ts
Comment (InnaAtanasova) [inline apps/docs/src/app/core/component-docs/card/examples/list-card/card-list-example.component.ts:34]: why do we have ChangeDetectorRef in the documentation part?
Not all applications consuming our library are using onpush strategy. The code should work with and without it

File: apps/docs/src/app/core/component-docs/card/examples/bar-chart-list-card/card-bar.component.scss
Comment (InnaAtanasova) [inline apps/docs/src/app/core/component-docs/card/examples/bar-chart-list-card/card-bar.component.scss:26]: Please stop adding random classes with random names. There are thousands of documentation examples with custom css from where you can see how we name the classes

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]
if you are touching something in the `lib` folder it's no longer a `docs` PR

**Comment by InnaAtanasova** [comment]
missing the keyword `fixes` or `closes` in the PR template in front of the issues that this PR is addressing. This will not close the issues after the PR is merged

File: apps/docs/src/app/core/component-docs/card/examples/bar-chart-list-card/card-bar.component.ts
Comment (DeepakSap14) [reply apps/docs/src/app/core/component-docs/card/examples/bar-chart-list-card/card-bar.component.ts]: fixed to displayChange = false

File: apps/docs/src/app/core/component-docs/card/card-docs.component.html
Comment (DeepakSap14) [reply apps/docs/src/app/core/component-docs/card/card-docs.component.html]: using `fd-docs-section-title` and `description` both now.

File: libs/core/src/lib/card/card.component.scss
Comment (DeepakSap14) [reply libs/core/src/lib/card/card.component.scss:18]: at line 19, variable with value is used.
anyway, We have to move these styles to fundamental-styles.
will raise a different PR in fundamental-styles.

File: libs/core/src/lib/card/card.component.scss
Comment (DeepakSap14) [reply libs/core/src/lib/card/card.component.scss:22]: not yet.
will raise a different PR in fundamental-styles to move all classes in this .scss file.

File: apps/docs/src/app/core/component-docs/card/examples/list-card/card-list-example.component.ts
Comment (DeepakSap14) [reply apps/docs/src/app/core/component-docs/card/examples/list-card/card-list-example.component.ts]: removed ChangeDetectorRef

File: apps/docs/src/app/core/component-docs/card/examples/bar-chart-list-card/card-bar.component.scss
Comment (DeepakSap14) [reply apps/docs/src/app/core/component-docs/card/examples/bar-chart-list-card/card-bar.component.scss]: corrected name now.

**Comment by DeepakSap14** [comment]
> missing the keyword `fixes` or `closes` in the PR template in front of the issues that this PR is addressing. This will not close the issues after the PR is merged

These issues are raised by `"design team"`. They will verify and close it. We have instructions to not close these defects by ourself.

File: apps/docs/src/app/core/component-docs/card/examples/link-list-card/card-link-list-example.component.scss
Comment (Betrozov) [inline apps/docs/src/app/core/component-docs/card/examples/link-list-card/card-link-list-example.component.scss:2]:
```suggestion
    background-color: #f7f7f7;
    background-color: var(--sapBackgroundColor, #f7f7f7);
```

File: libs/core/src/lib/card/card.component.scss
Comment (droshev) [reply libs/core/src/lib/card/card.component.scss:18]: @DeepakSap14 
There are 2 problems:
- using a hardcoded color value
- introducing css rules like this in ngx.

you can introduce this in styles(without hardcoded color value) and bring it from there. We are trying to reverse this practice.

**Review by droshev — CHANGES_REQUESTED** [review]
see my comment

**Comment by DeepakSap14** [comment]
i will be first raising a style PR. will wait for that PR to merge, then will come to this PR. if process takes longer, i may close this PR and open another PR in future.

**Comment by DeepakSap14** [comment]
Raised PR https://github.com/SAP/fundamental-styles/pull/2236 for moving styles. Styles related for card footer does not need to move but used bar component to create card footer.
This PR is blocked by https://github.com/SAP/fundamental-styles/pull/2236.

File: libs/core/src/lib/card/card.component.scss
Comment (InnaAtanasova) [reply libs/core/src/lib/card/card.component.scss:18]: @DeepakSap14 I apologize for the hard-coded color, I am used to seeing 6 letter code for css variables like #cccccc and didn't pay attention to the next line of code.
@droshev it's actually correct, he is adding the fallback for IE11 support.

**Comment by droshev** [comment]
@DeepakSap14 is it possible for you to split this PR per ticket? It is not a great practice to solve 6 issues in one PR

**Comment by droshev** [comment]
> Raised PR [SAP/fundamental-styles#2236](https://github.com/SAP/fundamental-styles/pull/2236) for moving styles. Styles related for card footer does not need to move but used bar component to create card footer.
> This PR is blocked by [SAP/fundamental-styles#2236](https://github.com/SAP/fundamental-styles/pull/2236).

not blocked anymore

**Comment by InnaAtanasova** [comment]
> > missing the keyword `fixes` or `closes` in the PR template in front of the issues that this PR is addressing. This will not close the issues after the PR is merged
> 
> These issues are raised by `"design team"`. They will verify and close it. We have instructions to not close these defects by ourself.

This is not something the "design team" can make as a decision for the library. Add them as reviewers if their approval is required, but the issues should be closed as soon as the PR is merged and it should be done automatically.

File: apps/docs/src/app/core/component-docs/card/examples/card-kpi-example.component.html
Comment (InnaAtanasova) [inline apps/docs/src/app/core/component-docs/card/examples/card-kpi-example.component.html:10]: Why do you introduce `class="fd-numeric-content__kpi--positive"`? We shouldn't have fundamental classes exposed like this, that's why we create directives

File: apps/docs/src/app/core/component-docs/fixed-card-layout/examples/default/fixed-card-layout-examples.component.html
Comment (InnaAtanasova) [inline apps/docs/src/app/core/component-docs/fixed-card-layout/examples/default/fixed-card-layout-examples.component.html:52]: same as above

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]

**Comment by InnaAtanasova** [comment]
It's not really clear what this PR is fixing by the title, please add more meaningful title. As I mentioned 100+ times, this goes in the release notes and it should be clear for the user what exactly was fixed.

File: apps/docs/src/app/core/component-docs/card/examples/card-kpi-example.component.html
Comment (DeepakSap14) [reply apps/docs/src/app/core/component-docs/card/examples/card-kpi-example.component.html]: i created input property in `card-kpi-value.directive.ts` to assign class name.

**Review by Betrozov — APPROVED** [review]

**Comment by Betrozov** [comment]
please fix Travis and netlify

File: libs/core/src/lib/card/kpi/card-kpi-value.directive.ts
Comment (InnaAtanasova) [inline libs/core/src/lib/card/kpi/card-kpi-value.directive.ts:19]: `fdType` is not a good name for this property. Something like `kpiType` or `kpiStatus` or just `status` would be more self-explanatory.

**Comment by InnaAtanasova** [comment]
Please fix Travis

File: libs/core/src/lib/card/kpi/card-kpi-value.directive.ts
Comment (DeepakSap14) [reply libs/core/src/lib/card/kpi/card-kpi-value.directive.ts]: renamed property to status.

**Comment by DeepakSap14** [comment]
@daisy-yovcheva @pierrom @yukioht 
please use this link https://deploy-preview-5015--fundamental-ngx.netlify.app to verify, if issue is fixed or not.
That will help us in closing the issue.

**Review by KevinOkamoto — APPROVED** [review]

File: apps/docs/src/app/core/component-docs/card/examples/card-example.component.html
Comment (InnaAtanasova) [inline apps/docs/src/app/core/component-docs/card/examples/card-example.component.html:26]: why do you need a wrapper div? you can apply a class directly on the fd-card

File: apps/docs/src/app/core/component-docs/card/examples/card-example.component.scss
Comment (InnaAtanasova) [inline apps/docs/src/app/core/component-docs/card/examples/card-example.component.scss:6]: so if user wants to have an image they have to add this margin-bottom?

File: apps/docs/src/app/core/component-docs/card/examples/card-example.component.html
Comment (DeepakSap14) [reply apps/docs/src/app/core/component-docs/card/examples/card-example.component.html]: removed extra div

File: apps/docs/src/app/core/component-docs/card/examples/card-example.component.scss
Comment (DeepakSap14) [reply apps/docs/src/app/core/component-docs/card/examples/card-example.component.scss]: image takes extra space at bottom.
have removed it now. as overriding is not required.

**Review by sKudum — APPROVED** [review]
LGTM

File: libs/core/src/lib/card/kpi/card-kpi-value.directive.ts
Comment (InnaAtanasova) [inline libs/core/src/lib/card/kpi/card-kpi-value.directive.ts:32]: Numeric content is a separate component that is not connected to Card component. It can be used to construct an analytical card but it can also be part of any other component. I don't understand why it's part of Cards.

**Review by InnaAtanasova — CHANGES_REQUESTED** [review]
Screen Shot 2021-05-10 at 11 22 50 AM

Inspecting the markup I see `analytical-card` class. I don't think it should be there.

**Review by mikerodonnell89 — CHANGES_REQUESTED** [review]
I think the issue with the table card border hasn't really been fixed here, true the gap is gone but the border surrounding the table appears to be wider than what it should be.  I think the problem is actually that the table header and the table body have borders that don't need to be there because they should simply be coming from the card.  This fix should also come from the fundamental-styles library

**Comment by DeepakSap14** [comment]
> There is a gap underneath the image and above the card title, inspecting the element i cannot tell where this is coming from (i am not talking about the issue described here: [SAP/fundamental-styles#2348](https://github.com/SAP/fundamental-styles/issues/2348))
> 
> Screen Shot 2021-05-11 at 3 03 47 PM

If you see this link https://stackoverflow.com/questions/5804256/image-inside-div-has-extra-space-below-the-image.
It is default behaviour of a image. for fixing it, we have to use css style on document side. which was asked to remove earlier https://github.com/SAP/fundamental-ngx/pull/5015#discussion_r626618390

**Comment by DeepakSap14** [comment]
I will be separating each issue in different PR's because each issue has different types of conflicts, stated below.
1. for #3928, first have to make changes in table component in styles repo, then that need to be brought to Ngx.
2. for #4710, It is documentation change, will keep a different PR. Image takes extra space at bottom, either we need to include it in styles, or need to have styles in documentation.
3. for #4243, KPI change has become more that anticipated. May need to refactor card to remove `analytical-card` classes.

I am closing this PR, and will address these issues in separate PR's. if any issue please let me know.

**Pull request closed without merging**
